packages/twenty-front/src/modules/object-record/record-index/components/RecordIndexBoardContainerEffect.tsx

```diff
@@ -98,5 +108,14 @@ export const RecordIndexBoardContainerEffect = ({
     setContextMenuEntries?.();
   }, [setActionBarEntries, setContextMenuEntries]);
 
+  useEffect(() => {
+    if (currentView?.isCompact !== isCompactModeActive) {
+      setIsCompactModeActive(!!currentView?.isCompact);
+    }
+    // We don't want to re-run this effect when the state changes.
+    // We only want to make sure that the state reflects the current view.
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
We should not try to block react re-rendering. It's an effect, it is not a problem :)